test/test_nested_tensor_masking.py

```diff
@@ -1,350 +1,976 @@
+                              [[True]]])
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask3)
+        TestCase.assertEqual(self, extected_nt1, res_nt)
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask3, nested_dim=2)
+        TestCase.assertEqual(self, extected_nt2, res_nt)
+
+        # Mask dim 2 
+        mask2 = torch.tensor([[True],
+                              [True],
+                              [True]])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask2)
+        TestCase.assertEqual(self, extected_nt1, res_nt)
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask2, nested_dim=2)
+        TestCase.assertEqual(self, extected_nt2, res_nt)
+
+        # Mask dim 1
+        mask1 = torch.tensor([True, True, True])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask1)
+        TestCase.assertEqual(self, extected_nt1, res_nt)
+        
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask1, nested_dim=2)
+        TestCase.assertEqual(self, extected_nt2, res_nt)
+
+    def test_ntftm_single_tensor_all_false_mask(self): 
+        tensor = torch.tensor([[1]])
+        mask = torch.tensor([False])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        TestCase.assertEqual(self, res_nt, nt.nested_tensor([]))
+
+        tensor = torch.tensor([[1, 2, 3]])
+        mask = torch.tensor([False])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        TestCase.assertEqual(self, res_nt, nt.nested_tensor([]))
+
+    def test_ntftm_multi_tensor_all_false_mask(self):
+        tensor = torch.tensor([[[1], [2], [3]]])
+        mask = torch.tensor([False])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        TestCase.assertEqual(self, res_nt, nt.nested_tensor([]))
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=2)
+        TestCase.assertEqual(self, res_nt, nt.nested_tensor([]))
+
+        mask = torch.tensor([False, False, False])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        TestCase.assertEqual(self, res_nt, nt.nested_tensor([]))
+
+        mask = torch.tensor([[False], [False], [False]])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        TestCase.assertEqual(self, res_nt, 
+                         nt.nested_tensor([
+                             torch.tensor([], dtype=tensor.dtype)
+                         ]))
+
+    def test_ntftm_multi_tensor_all_false_mask2(self):
+        tensor = torch.tensor([[[1], [2], [3]]])
+        mask = torch.tensor([[[False], [False], [False]]])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        TestCase.assertEqual(self, res_nt, 
+                         nt.nested_tensor([
+                            torch.empty((3, 0), dtype=tensor.dtype)
+                         ]))
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=2)
```

Comment:
Could do nested_dim=3 again for error testing but this time with all False

Reply:
This applies to the other cases as well. I assume there's a reason you don't want to do that.